src/factory.js

```diff
@@ -238,6 +239,11 @@ module.exports = ({ Component, createElement, PropTypes }) => {
                 return then.call(newState, setState);
               }
 
+              Object.keys(newState).forEach(key => {
+                if (!(key in state)) {
```

Comment:
It's better to use [`Object.prototype.hasOwnProperty.call`](https://devdocs.io/javascript/global_objects/object/hasownproperty) instead of `in` otherwise you might have false positives (for inherited properties).